app/screens/auth/UnlockWallet.tsx

```diff
@@ -84,16 +87,63 @@ const GrayText = styled.div`
   color: ${smColors.disabledGray};
 `;
 
+// TODO: Get rid from code duplication
+const AccItem = styled.div<{ isInDropDown: boolean }>`
+  width: 100%;
+  padding: 5px;
+  line-height: 17px;
+  font-size: 13px;
+  text-transform: uppercase;
+  color: ${smColors.black};
+  cursor: inherit;
+  ${({ isInDropDown }) => isInDropDown && `opacity: 0.5; border-bottom: 1px solid ${smColors.disabledGray};`}
+  &:hover {
+    opacity: 1;
+    color: ${smColors.darkGray50Alpha};
+  }
+`;
+
 const UnlockWallet = ({ history, location }: AuthRouterParams) => {
   const [password, setPassword] = useState('');
   const [isWrongPassword, setWrongPassword] = useState(false);
   const [showLoader, setShowLoader] = useState(false);
 
+  const walletFiles = useSelector(listWalletFiles);
+  const [selectedWalletIndex, setSelectedWalletIndex] = useState(getIndexOfLastSelectedWalletPath(walletFiles));
+
   const isWalletOnlyMode = useSelector(isWalletOnly);
   const isDarkMode = useSelector((state: RootState) => state.ui.isDarkMode);
   const dispatch: AppThDispatch = useDispatch();
   const chevronIcon = isDarkMode ? chevronRightWhite : chevronRightBlack;
 
+  useEffect(() => {
+    // Ensure that we had loaded wallet files
+    dispatch(readWalletFiles());
+  }, [dispatch]);
```

Comment:
```suggestion
  }, []);
```
if we want to insure, only on the first render , we don't need dispatch here, and can simplify it . up to u

Reply:
Sounds reasonable.
Meanwhile, in this case, I will have to turn off the eslint `react-hooks/exchaustive-deps` rule, and this is not recommended. Moreover, the `dispatch` function should not be changed over time and should have the same reference. So it should not be a problem 🤔